rust/main/hyperlane-base/src/cache/mod.rs

```diff
@@ -0,0 +1,34 @@
+mod cache_types;
+mod error;
+mod moka;
+
+pub use cache_types::*;
+pub use error::*;
+pub use moka::*;
+
+use async_trait::async_trait;
+use hyperlane_core::H256;
+use serde::{de::DeserializeOwned, Serialize};
+
+/// Hyperlane Cache Interface
+#[async_trait]
+pub trait HyperlaneCache: Send + Sync {
```

Comment:
I wonder if descriptive name will work better here like `ContractMethodCallCache`?

Reply:
As discussed, should be more usage/feature specific naming, so I renamed it to `FunctionCallCache` since not all calls which are made are contract calls.

[Fix](https://github.com/Mantas-M/hyperlane-monorepo/commit/3a0c126000a14ec47acdf1bdc4752e551362ecba)